net/api/service.go

```diff
@@ -58,18 +57,85 @@ func (s *Service) GetHostID(_ context.Context, _ *pb.GetHostIDRequest) (*pb.GetH
 	}, nil
 }
 
+type remoteIdentity struct {
+	pk     thread.PubKey
+	server pb.API_GetTokenServer
+}
+
+func (i *remoteIdentity) Sign(msg []byte) ([]byte, error) {
+	if err := i.server.Send(&pb.GetTokenReply{
+		Payload: &pb.GetTokenReply_Challenge{
+			Challenge: msg,
+		},
+	}); err != nil {
+		return nil, err
+	}
+
+	req, err := i.server.Recv()
+	if err != nil {
+		return nil, err
+	}
+	var sig []byte
+	switch payload := req.Payload.(type) {
+	case *pb.GetTokenRequest_Signature:
+		sig = payload.Signature
+	default:
+		return nil, status.Error(codes.InvalidArgument, "Signature is required")
+	}
+	return sig, nil
+}
+
+func (i *remoteIdentity) GetPublic() thread.PubKey {
+	return i.pk
+}
+
+func (s *Service) GetToken(server pb.API_GetTokenServer) error {
```

Comment:
On the remote-side:
1. Save the received pubkey (it gets included in the JWT)
2. Send back a random challenge (server-side nonce). Again, I don't think we need a client-side nonce here since there can be no man in the middle (SSL). However, it would be super easy to add in the original message send to the remote.
3. `net.GetToken` takes an implementation of `Identity`. We can build one with the bidi server and sent pubkey, and let that method finish the job of minting a token.